pkg/controller/cluster/cluster_controller.go

```diff
@@ -594,37 +571,38 @@ func buildNewNodeGroup(cluster *clusteroperator.Cluster, clusterNodeGroup cluste
 				},
 			},
 		},
-		Spec: clusteroperator.NodeGroupSpec{
-			NodeType: nodeType,
-			Size:     clusterNodeGroup.Size,
-		},
+		Spec: machineSetSpec,
 	}
 }
 
-func getNamePrefixForMasterNodeGroup(cluster *clusteroperator.Cluster) string {
-	return fmt.Sprintf("%s-master-", cluster.Name)
-}
-
-func getNamePrefixForComputeNodeGroup(cluster *clusteroperator.Cluster, nodeGroupName string) string {
-	return fmt.Sprintf("%s-compute-%s-", cluster.Name, nodeGroupName)
+func getNamePrefixForMachineSet(cluster *clusteroperator.Cluster, machineSet clusteroperator.ClusterMachineSet) string {
+	nodeTypePrefix := ""
+	switch machineSet.Spec.NodeType {
+	case clusteroperator.NodeTypeMaster:
+		nodeTypePrefix = "master"
+	case clusteroperator.NodeTypeCompute:
+		nodeTypePrefix = "compute"
+	}
+	return fmt.Sprintf("%s-%s-%s-", cluster.Name, nodeTypePrefix, machineSet.Name)
```

Comment:
The nodeTypePrefix is not necessary. It is sufficient for the machine set name prefix to be `fmt.Sprintf("%s-%s-", cluster.Name, machineSet.Name)`. The "compute" was necessary before because you could have conceivably had a compute node that the user had named master.